src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -92,135 +97,135 @@ parseAMM(Json::Value const& params, Json::Value& jvResult)
-        return std::nullopt;
-    }
-
-    return uNodeIndex;
+    return parseObjectID(params, fieldName, "hex string");
 }
 
-static std::optional<uint256>
-parseCredential(Json::Value const& cred, Json::Value& jvResult)
+static Expected<uint256, Json::Value>
+parseCredential(Json::Value const& cred, Json::StaticString const& fieldName)
 {
-    if (cred.isString())
+    if (!cred.isObject())
     {
-        uint256 uNodeIndex;
-        if (!uNodeIndex.parseHex(cred.asString()))
-        {
-            jvResult[jss::error] = "malformedRequest";
-            return std::nullopt;
-        }
-        return uNodeIndex;
+        return parseObjectID(cred, fieldName);
     }
 
-    if ((!cred.isMember(jss::subject) || !cred[jss::subject].isString()) ||
-        (!cred.isMember(jss::issuer) || !cred[jss::issuer].isString()) ||
-        (!cred.isMember(jss::credential_type) ||
-         !cred[jss::credential_type].isString()))
-    {
-        jvResult[jss::error] = "malformedRequest";
-        return std::nullopt;
-    }
+    auto const subject =
+        requiredAccountID(cred, jss::subject, "malformedRequest");
+    if (!subject)
+        return Unexpected(subject.error());
 
-    auto const subject = parseBase58<AccountID>(cred[jss::subject].asString());
-    auto const issuer = parseBase58<AccountID>(cred[jss::issuer].asString());
-    auto const credType = strUnHex(cred[jss::credential_type].asString());
+    auto const issuer =
+        requiredAccountID(cred, jss::issuer, "malformedRequest");
+    if (!issuer)
+        return Unexpected(issuer.error());
 
-    if (!subject || subject->isZero() || !issuer || issuer->isZero() ||
-        !credType || credType->empty())
-    {
-        jvResult[jss::error] = "malformedRequest";
-        return std::nullopt;
-    }
+    auto const credType = requiredHexBlob(
+        cred,
+        jss::credential_type,
+        maxCredentialTypeLength,
+        "malformedRequest");
+    if (!credType)
+        return Unexpected(credType.error());
 
     return keylet::credential(
                *subject, *issuer, Slice(credType->data(), credType->size()))
         .key;
 }
 
-static STArray
+static Expected<uint256, Json::Value>
+parseDelegate(Json::Value const& params, Json::StaticString const& fieldName)
+{
+    if (!params.isObject())
+    {
+        return parseObjectID(params, fieldName);
+    }
+
+    auto const account =
+        requiredAccountID(params, jss::account, "malformedAddress");
+    if (!account)
+        return Unexpected(account.error());
+
+    auto const authorize =
+        requiredAccountID(params, jss::authorize, "malformedAddress");
+    if (!authorize)
+        return Unexpected(authorize.error());
+
+    return keylet::delegate(*account, *authorize).key;
+}
+
+static Expected<STArray, Json::Value>
 parseAuthorizeCredentials(Json::Value const& jv)
 {
+    if (!jv.isArray())
+        return invalidFieldError(
+            "malformedAuthorizedCredentials",
```

Comment:
I highly recommend putting this to anonymous namespace as StaticString and pass the variables instead of hardcoding some string.